### ISSUE TITLE
pokeemerald.elf white screen

### ISSUE THREAD

**Issue by vblm12** [mgba-emu/mgba#1450]
Playing around with the pokemon decomps, I get a white screen if I load the elf for debugging, including the vanilla pokeemerald build. I see this in both mgba-qt 0.7.2 on Arch and git master. I made sure to try building with `-DUSE_ELF=ON` since I don't know if it was enabled by default.

With max log level and duplicate lines removed I get:
```
[GAME ERROR] GBA I/O: Write to read-only I/O register: 006
[GAME ERROR] GBA I/O: Read from write-only I/O register: 0DC
[GAME ERROR] GBA I/O: Read from write-only I/O register: 0C4
[GAME ERROR] GBA I/O: Read from write-only I/O register: 0D0
[GAME ERROR] GBA Memory: Bad BIOS Load8: 0x0000000A
[GAME ERROR] GBA Memory: Bad memory Store16: 0x00000000
[GAME ERROR] GBA Memory: Bad memory Store16: 0x00000002
[GAME ERROR] GBA Memory: Bad memory Store16: 0x00000004
[GAME ERROR] GBA Memory: Bad memory Store16: 0x00000006
[GAME ERROR] GBA Memory: Bad memory Store16: 0x00000008
[GAME ERROR] GBA Memory: Bad memory Store16: 0x0000000A
[GAME ERROR] GBA Memory: Bad memory Store16: 0x0000000C
[GAME ERROR] GBA Memory: Bad memory Store16: 0x0000000E
[GAME ERROR] GBA Memory: Bad memory Store16: 0x00000010
[GAME ERROR] GBA Memory: Bad memory Store16: 0x00000012
[GAME ERROR] GBA Memory: Bad memory Store16: 0x00000014
[GAME ERROR] GBA Memory: Bad memory Store16: 0x00000016
[GAME ERROR] GBA Memory: Bad memory Store16: 0x00000018
[GAME ERROR] GBA Memory: Bad memory Store16: 0x0000001A
[GAME ERROR] GBA Memory: Bad memory Store16: 0x0000001C
[GAME ERROR] GBA Memory: Bad memory Store16: 0x0000001E
[GAME ERROR] GBA Memory: Bad memory Store16: 0x00000020
[GAME ERROR] GBA Memory: Bad memory Store16: 0x00000022
[GAME ERROR] GBA Memory: Bad memory Store16: 0x00000024
[GAME ERROR] GBA Memory: Bad memory Store16: 0x00000026
[GAME ERROR] GBA Memory: Bad memory Store16: 0x00000028
[GAME ERROR] GBA Memory: Bad memory Store16: 0x0000002A
[GAME ERROR] GBA Memory: Bad memory Store16: 0x0000002C
[GAME ERROR] GBA Memory: Bad memory Store16: 0x0000002E
[GAME ERROR] GBA Memory: Bad memory Store16: 0x00000030
[GAME ERROR] GBA Memory: Bad memory Store16: 0x00000032
[GAME ERROR] GBA Memory: Bad memory Store16: 0x00000034
[GAME ERROR] GBA Memory: Bad memory Store16: 0x00000036
[GAME ERROR] GBA Memory: Bad cartridge Store16: 0x080000C8
[GAME ERROR] GBA Memory: Bad cartridge Store16: 0x080000C4
[GAME ERROR] GBA Memory: Bad cartridge Store16: 0x080000C6
[GAME ERROR] GBA Savedata: Bad flash write: 0x5555 = 0xf0
```

Is pokeemerald doing something wrong in the way it's building the elf?

**Comment by endrift** [comment]
This happens because the ELF doesn't include the GBA header, so mGBA can't detect the right overrides to use. In theory this can be fixed by looking up the AgbLibFlashVersion symbol and reading its value. However, this would require loading symbols whenever an ELF is loaded, which I don't do at the moment.

**Comment by PikalaxALT** [comment]
As I mentioned in #1481, you can compile my version of the `gbafix` utility from [here][h] and use that to set the header on the ELF in place.

[h]: /pikalaxalt/pokeemerald/tree/modern_gcc/tools/gbafix

**Comment by endrift** [comment]
This should have been fixed in 85b4bb14004e9fb2c6bbb031d35b8a6000da6068.
